src/modules/Dropdown/Dropdown.js

```diff
@@ -373,63 +419,109 @@ export default class Dropdown extends Component {
     e.stopPropagation()
     const { search } = this.props
     const { open } = this.state
-    const newValue = e.target.value
+    const newQuery = e.target.value
 
     // open search dropdown on search query
-    if (search && newValue && !open) this.open()
+    if (search && newQuery && !open) this.open()
 
     this.setState({
       selectedIndex: 0,
-      searchQuery: newValue,
+      searchQuery: newQuery,
     })
   }
 
   // ----------------------------------------
   // Getters
   // ----------------------------------------
 
-  getOptions = () => {
-    const { options, search } = this.props
-    const { searchQuery } = this.state
+  getMenuOptions = () => {
+    const { multiple, options, search } = this.props
+    const { searchQuery, value } = this.state
 
-    if (!search || !searchQuery) return options
+    let filteredOptions = options
 
-    const re = new RegExp(_.escapeRegExp(searchQuery), 'i')
-    return _.filter(options, (opt) => re.test(opt.text))
+    // filter out active options
+    if (multiple) {
+      filteredOptions = _.filter(filteredOptions, opt => !_.includes(value, opt.value))
+    }
+
+    // filter by search query
+    if (search && searchQuery) {
+      const re = new RegExp(_.escapeRegExp(searchQuery), 'i')
+      filteredOptions = _.filter(filteredOptions, (opt) => re.test(opt.text))
+    }
+
+    return filteredOptions
   }
 
-  getSelectedValue = () => {
-    debug('getSelectedValue()')
+  getSelectedItemValue = () => {
     const { selectedIndex } = this.state
-    const options = this.getOptions()
+    const options = this.getMenuOptions()
 
     return _.get(options, `[${selectedIndex}].value`)
   }
 
-  getItemIndexByValue = (value) => {
-    debug('getItemIndexByValue()')
-    const options = this.getOptions()
+  getItemByValue = (value) => {
+    const { options } = this.props
+    return _.find(options, { value })
+  }
+
+  getMenuItemIndexByValue = (value) => {
+    const options = this.getMenuOptions()
 
-    return _.findIndex(options, ['value', value])
+    return _.findIndex(options, opt => opt.value === value)
```

Comment:
Previous places you are using shorthand (with `find`):

`return _.findIndex(options, { value })`